reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java

```diff
@@ -713,9 +741,34 @@ public final HttpServer secure(Consumer<? super SslProvider.SslContextSpec> sslP
 	 * @return a new {@link HttpServer}
 	 */
 	public final HttpServer secure(SslProvider sslProvider) {
+		return secure(sslProvider, false);
+	}
+
+	/**
+	 * Applies an SSL configuration via the passed {@link SslProvider}.
+	 * <p>
+	 * If {@link SelfSignedCertificate} needs to be used, the sample below can be
+	 * used. Note that {@link SelfSignedCertificate} should not be used in production.
+	 * <pre>
+	 * {@code
+	 *     SelfSignedCertificate cert = new SelfSignedCertificate();
+	 *     SslContextBuilder sslContextBuilder =
+	 *             SslContextBuilder.forServer(cert.certificate(), cert.privateKey());
+	 *     secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder), true);
+	 * }
+	 * </pre>
+	 *
+	 * @param sslProvider         The provider to set when configuring SSL
+	 * @param redirectHttpToHttps true enables redirecting HTTP to HTTPS on the same port.
+	 *                            This configuration is applicable only for HTTP/1.x.
```

Comment:
```suggestion
	 * @param redirectHttpToHttps true enables redirecting HTTP to HTTPS by changing the
	 *                            scheme only but otherwise leaving the port the same.
	 *                            This configuration is applicable only for HTTP 1.x.
```

Reply:
updated.